src/nvim/ex_cmds.lua

```diff
@@ -38,400 +38,544 @@ module.flags = {
   PREVIEW = PREVIEW,
 }
 
+--- See `src/nvim/ex_cmds_defs.h` for the meaning of each field
+--- @class vim.ExCmd
+--- @field command string name of Ex command, e.g., 'edit'
+--- @field short_command string shorter name without the optional part, e.g., 'e' for :e[dit].
+--- @field desc string a brief description, see |ex-cmd-index|
+--- @field enum? string
+--- @field flags integer
+--- @field addr_type string
+--- @field func string
+--- @field removed? boolean deprecated or removed in neovim, should not appear in the docs
+
 -- The following table is described in ex_cmds_defs.h file.
+--- @type vim.ExCmd[]
 module.cmds = {
   {
     command = 'append',
+    short_command = 'a',
+    desc = [[append text]],
     flags = bit.bor(BANG, RANGE, ZEROR, TRLBAR, CMDWIN, LOCK_OK, MODIFY),
     addr_type = 'ADDR_LINES',
     func = 'ex_append',
   },
   {
     command = 'abbreviate',
+    short_command = 'ab',
+    desc = [[enter abbreviation]],
     flags = bit.bor(EXTRA, TRLBAR, NOTRLCOM, CTRLV, CMDWIN, LOCK_OK),
     addr_type = 'ADDR_NONE',
     func = 'ex_abbreviate',
   },
   {
     command = 'abclear',
+    short_command = 'abc',
+    desc = [[remove all abbreviations]],
     flags = bit.bor(EXTRA, TRLBAR, CMDWIN, LOCK_OK),
     addr_type = 'ADDR_NONE',
     func = 'ex_abclear',
   },
   {
     command = 'aboveleft',
+    short_command = 'abo',
+    desc = [[make split window appear left or above]],
     flags = bit.bor(NEEDARG, EXTRA, NOTRLCOM),
     addr_type = 'ADDR_NONE',
     func = 'ex_wrongmodifier',
   },
   {
     command = 'all',
+    short_command = 'al',
+    desc = [[open a window for each file in the argument list]],
     flags = bit.bor(BANG, RANGE, COUNT, TRLBAR),
     addr_type = 'ADDR_OTHER',
     func = 'ex_all',
   },
   {
     command = 'amenu',
+    short_command = 'am',
+    desc = [[enter new menu item for all modes]],
```

Comment:
eval.lua uses `desc` for the full documentation. So we should keep that in mind and use a different field for the "short" doc?

```suggestion
    shortdesc = [[enter new menu item for all modes]],
```

Reply:
Or we could generate the "short desc" for the index from the `desc`, by using the first X chars.

I think that's better, it avoids an extra field that must be maintained.

Reply:
That would be ideal but currently the full `desc` docs for many commands don't necessarily start with the short desc (unless we update the docs to be written so) -- actually quite different (though semantically same). We'll have slightly different docs (short and full desc) than Vanilla vim's, but this would be fine.

`:amenu`:

>  The ":amenu" command can be used to define menu entries for all modes at once,

Reply:
> Or we could generate the "short desc" for the index from the `desc`, by using the first X chars.
> 
> I think that's better, it avoids an extra field that must be maintained.

I think that will make maintenance even harder.

Reply:
> unless we update the docs to be written so

I don't think we need to be tied to how "official vim" documents these commands. We can do whatever makes sense for us.

Reply:
ok, for now let's just use `desc` as the short description. We aren't moving the long descs into this file any time soon.